.github/workflows/test.yml

```diff
@@ -58,3 +58,7 @@ jobs:
     - name: run regression tests
       run: |
        make test
+    - name: code coverage
+      run: |
+       PYTHON="coverage run" make test_unittest
+       make coverage_report
```

Comment:
I dislike such way to run collect / report coverage. Can it be as easy as `make coverage` or `make test COVERAGE=1`?

Reply:
Could you explain why you dislike this way? Why running `make test COVERAGE=1` is better than `make coverage_report`?
Below you suggested pushing code coverage data to coveralls.io because it is "more useful comparing to just numbers".
Why it is important how we run code coverage collecting process if we plan to use it in GH actions and not manually?

Reply:
> Could you explain why you dislike this way?

It is not what I can remember or find somewhere in the Makefile.

> Why running `make test COVERAGE=1` is better than `make coverage_report`?

The first one doing everything (collecting + reporting), not just reporting.

> Why it is important how we run code coverage collecting process if we plan to use it in GH actions and not manually?

The ability to run manually is what I expect from a project that has a Makefile in the root directory. It is useful for debugging (say, looking why a test does not cover a particulal line of the code). I would not place rules that are solely for CI into the Makefile.

Reply:
now it is enough to execute `make coverage`

Reply:
Nice!